src/simplewallet/simplewallet.cpp

```diff
@@ -563,7 +563,7 @@ bool simple_wallet::viewkey(const std::vector<std::string> &args/* = std::vector
 {
   if (m_wallet->ask_password() && !get_and_verify_password()) { return true; }
   // don't log
-  if (m_wallet->key_on_device()) {
+  if (m_wallet->key_on_device() != hw::device::device_type::SOFTWARE) {
```

Comment:
Not sure if you're yet to change the last things, but this'd be a lot nicer IMHO with:

- keeping m_wallet->key_on_device()
- having bool key_on_device() const { return get_device_type() != SOFTWARE; }

Less diff spam, and it it behaves more like one expects.

Reply:
i agree - will do this & probably ```get_key_backing_type``` by tomorrow evening.